### PR TITLE
Add types declaration file with entry in package.json

### LINKED ISSUE #4: TypeScript type declaration file
I found your library from Reddit. If you want I can write a TypeScript type declaration file for this library so that TS developers can utilise the type-checking feature of the language as they use `whenipress`. Let me know if you are interested.

### PR DESCRIPTION
**Pull request by geopic**
Resolves #4. Let me know if there are any issues.

### CODE REVIEW

File: whenipress.d.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    type WhenIPressPlugin = Partial<{
+      mounted: (globalInstance: PendingKeyboardEventManager) => void;
+      bindingRegistered: (binding: string[], globalInstance: PendingKeyboardEventManager) => void;
+      bindingStopped: (binding: string[], globalInstance: PendingKeyboardEventManager) => void;
+      allBindingsStopped: (globalInstance: PendingKeyboardEventManager) => void;
+      beforeBindingHandled: (binding: string[], globalInstance: PendingKeyboardEventManager) => boolean | void;
+      afterBindingHandled: (binding: string[], globalInstance: PendingKeyboardEventManager) => void;
+    }>
+
+    type WhenIPressPluginOptions = Partial<{
```
Comment (lukeraymonddowning) [inline whenipress.d.ts]: I don't think we need this part. Its only referenced in the documentation and not used anywhere in the source code.

File: whenipress.d.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    class PendingKeyboardEvent {
+      constructor(manager: PendingKeyboardEventManager, ...keys: string[]);
+      then(handler: Handler): this;
+      do(handler: Handler): this;
+      run(handler: Handler): this;
+      whenReleased(handler: Handler): this;
+      once(): this;
+      twiceRapidly(timeout?: number): this;
+      stop(): void;
+      createKeyDownListener(handler: Handler): void;
```
Comment (lukeraymonddowning) [inline whenipress.d.ts]: This has been replaced with `createKeyDownHandler`

File: whenipress.d.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      removeKeyDownListener(): void;
+      createKeyUpListener(handler: Handler): void;
```
Comment (lukeraymonddowning) [inline whenipress.d.ts]: This has been replaced with `createKeyUpHandler`

File: whenipress.d.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      constructor(manager: PendingKeyboardEventManager, ...keys: string[]);
+      then(handler: Handler): this;
+      do(handler: Handler): this;
+      run(handler: Handler): this;
+      whenReleased(handler: Handler): this;
+      once(): this;
+      twiceRapidly(timeout?: number): this;
+      stop(): void;
+      createKeyDownListener(handler: Handler): void;
+      removeKeyDownListener(): void;
```
Comment (lukeraymonddowning) [inline whenipress.d.ts]: This has been removed, as children no longer register event listeners.

File: whenipress.d.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      createKeyUpListener(handler: Handler): void;
+      removeKeyUpListener(): void;
```
Comment (lukeraymonddowning) [inline whenipress.d.ts]: This has been removed, as children no longer register event listeners.

**Review by lukeraymonddowning — CHANGES_REQUESTED** [review]

File: whenipress.d.ts (hunk 1 above)
Comment (geopic) [reply whenipress.d.ts]: But this is for the developer using this library when they're writing their own plugin with individual hooks like `mounted`, `bindingRegistered` etc.

File: whenipress.d.ts (hunk 2 above)
Comment (geopic) [reply whenipress.d.ts]: Noted

File: whenipress.d.ts (hunk 4 above)
Comment (geopic) [reply whenipress.d.ts]: Noted

File: whenipress.d.ts (hunk 3 above)
Comment (geopic) [reply whenipress.d.ts]: Noted

File: whenipress.d.ts (hunk 5 above)
Comment (geopic) [reply whenipress.d.ts]: Noted

File: whenipress.d.ts (hunk 1 above)
Comment (lukeraymonddowning) [reply whenipress.d.ts]: Understood. Just wonder what the purpose of `urlsToSkip` and `skipAllUrls` is here though? That isn't a global plugin option, its just an example of options that a plugin owner might choose to include.

Am I missing something?

Comment (geopic) [reply whenipress.d.ts]: So they're entirely custom options defined by the developer using the library and they can just as well use anything for options like `{ foo: true, bar: false }` etc? Just need to be clear :sweat_smile:

Comment (lukeraymonddowning) [reply whenipress.d.ts]: Indeed. I probably need to clear that up in the documentation. Its just a unified area for providing configuration options to user's of your plugin.

Comment (geopic) [reply whenipress.d.ts]: Ok, cool. Thanks for clearing it up. I'll push a new commit now.

**Review by lukeraymonddowning — APPROVED** [review]
Awesome work! Thanks!

**Comment by lukeraymonddowning** [comment]
@all-contributors please add @geopic for code

**Comment by lukeraymonddowning** [comment]
@all-contributors please add @geopic for code

**Pull request merged**